daemon/api_test.go

```diff
@@ -2751,7 +2732,12 @@ func (s *apiSuite) TestLocalInstallSnapDeriveSideInfo(c *check.C) {
 		"timestamp":     time.Now().Format(time.RFC3339),
 	}, nil, "")
 	c.Assert(err, check.IsNil)
-	assertAdd(st, snapRev)
+
+	func() {
```

Comment:
because in the latter code because these helpers panic, you'll get  a panic about the lock state instead of the real interesting panic. In general  lock; do something; unlock is bad because of that even in the prod code though there we do get a bit more information I think because nothing is catching the panics